idea/src/org/jetbrains/kotlin/idea/inspections/RedundantUnitReturnTypeInspection.kt

```diff
@@ -43,6 +44,7 @@ class RedundantUnitReturnTypeInspection : AbstractKotlinInspection() {
                     if (!function.hasBlockBody()) {
                         val bodyExpression = function.bodyExpression
                         if (bodyExpression != null) {
+                            if (bodyExpression is KtThrowExpression) return
```

Comment:
It's not the correct way. `KtThrowException` is not the only one that has `Nothing` type, e.g. `null!!` has the same type. You should check `bodyExpression` type directly.

Reply:
Fixed